### DISCUSSION TITLE
Sum function in combination with appended  cards

### DISCUSSION THREAD

**Discussion by gacsphere** [neuefische/questions#274] [Web] [answered]
Hello everybody,

in my app I map over an array (initialNestingBoxes) to display all objects as cards in a cardlist. In each object I have a count-property, and I display the total amount of the count via a sum-function. It works so far. The total amount is 66 (33 + 33) and it is displayed.

Now I added a Form-component to create new cards (also with a count-property) and append them to the card list. Now the sum-function doesn't give back the correct total amount any more. 
When I append a new card via the form with `count: 1` the sum function appends `1` to `66` so that the new count is displayed as `661`. But it should be `67` (66 + 1). 

```JS
const initialNestingBoxes = [
  {
    id: nanoid(),
    date: "31.08.2021",
    time: "15:56 Uhr",
    latitude: 49.138844,
    longitude: 8.278068,
    boxnumber: 246,
    count: 33,
  },
  {
    id: nanoid(),
    date: "31.08.2021",
    time: "16:00 Uhr",
    latitude: 49.139855,
    longitude: 8.278925,
    boxnumber: 247,
    count: 33,
  },
];

export default function Home() {
  const [nestingBoxes, setNestingBoxes] = useState(initialNestingBoxes);

  function appendCard(date, time, latitude, longitude, boxnumber, count) {
    setNestingBoxes((nestingBoxes) => [
      ...nestingBoxes,
      {
        date,
        time,
        latitude,
        longitude,
        boxnumber,
        count,
        id: nanoid(),
      },
    ]);
  }

  const sum = nestingBoxes
    .map((nestingbox) => {
      return nestingbox.count;
    })
    .reduce((a, b) => a + b, 0);

  return (
    <div>
...
      <main>
        <h1>{sum} Fledermäuse insgesamt</h1>
        <Form onCreate={appendCard} />
        <Cards nestingBoxes={nestingBoxes} />
      </main>
    </div>
  );
}
`

Can someone help?

**Comment by NikRoe** [comment, marked as answer]
Hey there,

the issue you are currently facing is that you are trying to add a number with a string. 
The values in your initial array that lie under the key `cost` are of the type number. The new data you create via your form however, are of the type string. 
Values from an input field are always of the type string, regardless of assigning this input field `type="number"`.

What you need to do is convert the newly created `cost` from a string to a number for your `.reduce()` to work properly.

It could look similar to this:
```js
//...
setNestingBoxes((nestingBoxes) => [
      ...nestingBoxes,
      {
        date,
        time,
        latitude,
        longitude,
        boxnumber,
        count: Number(count),
        id: nanoid(),
      },
    ]);
//...
```

Hope this already help :)

-Niklas

**Reply by gacsphere** [reply]
Thank you Niklas!
That's it. 
:)
